### PR TITLE
NSBSD os support for VMBackup extension

### PR DESCRIPTION
**Pull request by remip2**
This patch adds support for NSBSD OS for the Backup extension.

I installed manually the patched extension and backups now end with success.

Issue #896

### CODE REVIEW

File: VMBackup/main/Utils/HandlerUtil.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def _parse_config(self, ctxt):
                 f.close()
                 waagent.SetFileContents(f.name,config['runtimeSettings'][0]['handlerSettings']['protectedSettings'])
                 cleartxt = None
-                cleartxt = waagent.RunGetOutput(self.patching.base64_path + " -d " + f.name + " | " + self.patching.openssl_path + " smime  -inform DER -decrypt -recip " + cert + "  -inkey " + pkey)[1]
+                cleartxt = waagent.RunGetOutput(self.patching.openssl_path + " base64 -d -A -in " + f.name + " | " + self.patching.openssl_path + " smime  -inform DER -decrypt -recip " + cert + "  -inkey " + pkey)[1]
```
Comment (soumyapattnaik) [inline VMBackup/main/Utils/HandlerUtil.py]: please have this code specific to NS-BSD. We dont want to break it for other supported OS

File: VMBackup/main/patch/__init__.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 import os
 import re
 import platform
+import traceback
```
Comment (soumyapattnaik) [inline VMBackup/main/patch/__init__.py]: this import is unwanted

File: VMBackup/main/handle.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
```
Comment (soumyapattnaik) [inline VMBackup/main/handle.sh]: please revert this to bash shell. For freeBSD we are suggesting customers to install bash. We have plans to move this to bourne shell in few days. It has to be tested with all the supported distros.

File: VMBackup/main/patch/__init__.py (hunk 2 above)
Comment (remip2) [reply VMBackup/main/patch/__init__.py]: the lib is used line 64, it's something i came across during debugging

File: VMBackup/main/handle.sh (hunk 3 above)
Comment (remip2) [reply VMBackup/main/handle.sh]: Unfortunately, NSBSD is not FreeBSD, Bash is not available and I can't add it. If you already plan to move to sh, I can wait and merge. At first look there is no bash specific syntax in the script.

File: VMBackup/main/Utils/HandlerUtil.py (hunk 1 above)
Comment (remip2) [reply VMBackup/main/Utils/HandlerUtil.py]: ok, a fix has been pushed

File: VMBackup/main/handle.sh (hunk 3 above)
Comment (soumyapattnaik) [reply VMBackup/main/handle.sh]: got it. I will merge the changes with bourne shell, as we plan to support bourne shell in next release

**Comment by remip2** [comment]
@soumyapattnaik thanks for the merge, do you have an idea when this version of vmbackup will be deployed live on Azure ? I tested today and this is still Microsoft.Azure.RecoveryServices__VMSnapshotLinux__1.0.9157.0 without the patch which is downloaded.

**Pull request merged**